src/app-layout/utils/use-focus-control.ts

```diff
@@ -18,6 +18,81 @@ export interface FocusControlState {
   loseFocus: () => void;
 }
 
+export interface FocusControlMultipleStates {
+  refs: Record<string, FocusControlRefs>;
+  setFocus: (params?: { force?: boolean; drawerId?: string }) => void;
+  loseFocus: () => void;
+}
+
+export function useMultipleFocusControl(
+  restoreFocus: boolean,
+  activeDrawersIds: Array<string>
+): FocusControlMultipleStates {
+  const isOpen = activeDrawersIds.length > 0;
+  const refs = useRef<Record<string, FocusControlRefs>>({});
+
+  activeDrawersIds.forEach(drawerId => {
```

Comment:
I tried implementing it using `useEffect` with `activeDrawersIds` as a dependency but ran into a race condition: the DOM elements weren't assigned to their refs after the first drawer opened, so setFocus received `undefined` value initially.